sdks/python/apache_beam/runners/worker/crossplatform_time.h

```diff
@@ -0,0 +1,55 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef BEAM_CROSSPLATFORM_TIME_H
+#define BEAM_CROSSPLATFORM_TIME_H
+
+#include <time.h>
+
+#ifdef _WIN32
+#include <windows.h>
+
+/**
+ * Alternative to POSIX clock_gettime that may be run on Windows platform. The clk_id parameter is
+ * ignored, and function always act as for CLOCK_MONOTONIC. Windows performance counter is used.
+ */
+int clock_gettime(int clk_id, struct timespec *tv) {
+    static LARGE_INTEGER counterFrequency = {0};
+    LARGE_INTEGER counterValue;
+
+    if (0 == counterFrequency.QuadPart) {
+        if (0 == QueryPerformanceFrequency(&counterFrequency)) {
+            /* System doesn't support performance counters. It's guaranteed to not happen
+            on systems that run Windows XP or later */
+            return -1;
```

Comment:
These error codes are never checked by the callers. That would seem to give incorrect results (especially if timespec is initialized with random data (e.g. on the stack) each time. 

Is there a simpler/more robust API we could call instead in this case, even if it is lower precision?

Reply:
Nice catch! :) 

The performance counter is the only API that is designed to measure execution time that I know. There is `GetSystemTime` and family with lower precision, which will act more like `CLOCK_REALTIME`. I see four solutions there:

1. Leave it as it is. I'm not sure if that's a real-life issue in our case, as the function is guaranteed to succeed on WinXP+ (?)
2. Switch to the less accurate `GetSystemTime`
3. Use `GetSystemTime` as a failover procedure if `PerformanceCounter` Failed. The downside here is that we will have different clk types (`MONOTHONIC`like and `REALTIME`like ) depending on the platform.
4. Write some fancy logic to merge the PerformanceCounter and GetSystemTime to have precise `REALTIME`like output and failover to less precise `GetSystemTime`.

What do you think?

Reply:
IMO, both 3 and 1 sounds fine. 3 might be slightly better. WinXP+ is practically all modern Windows versions.

Reply:
I've just pushed the third option